lib/xml_wrap.mli

```diff
@@ -30,5 +30,27 @@ module type T = sig
   val map : ('a -> 'b ) -> 'a tlist -> 'b tlist
 end
 
+module type T' = sig
+
+  type 'a t
+  val return : 'a -> 'a t
+
+  type ('a, 'b) ft
+
+  val fmap : ('a, 'b) ft -> 'a t -> 'b t
+
+  type 'a tlist
+  val nil : unit -> 'a tlist
+  val singleton : 'a t -> 'a tlist
+  val cons : 'a t -> 'a tlist -> 'a tlist
+  val append : 'a tlist -> 'a tlist -> 'a tlist
+  val map : ('a, 'b) ft -> 'a tlist -> 'b tlist
+
+end
+
 module NoWrap : T with type 'a t = 'a
                    and type 'a tlist = 'a list
+
+module NoWrap' : T' with type 'a t = 'a
+                     and type 'a tlist = 'a list
+                     and type ('a, 'b) ft = 'a -> 'b
```

Comment:
the type of NoWrap' is more precise but included in the type of NoWrap, so no need for the two version, just define the most precise one (with the equalities).